spec/helpers/http_mock/nginx_instance.lua

```diff
@@ -0,0 +1,74 @@
+local template_str = require "spec.helpers.http_mock.template"
+local pl_template = require "pl.template"
+local pl_path = require "pl.path"
+local pl_dir = require "pl.dir"
+local pl_file = require "pl.file"
+local pl_utils = require "pl.utils"
+local os = require "os"
+
+local print = print
+local error = error
+local assert = assert
+local ngx = ngx
+local io = io
+local shallow_copy = require "kong.tools.utils".shallow_copy
+
+local template = assert(pl_template.compile(template_str))
+local render_env = {ipairs = ipairs, pairs = pairs, error = error, }
+
+local http_mock = {}
+
+-- start a dedicate nginx for this mock
+function http_mock:start(error_on_exist)
+  local ok = (pl_path.mkdir(self.prefix))
+    and (pl_path.mkdir(self.prefix .. "/logs"))
+    and (pl_path.mkdir(self.prefix .. "/conf"))
+  if error_on_exist then assert(ok, "failed to create directory " .. self.prefix) end
+  
+  local render = assert(template:render(shallow_copy(self), render_env))
+  local conf_path = self.prefix .. "/conf/nginx.conf"
+  local conf_file = assert(io.open(conf_path, "w"))
+  assert(conf_file:write(render))
+  assert(conf_file:close())
+
+  local cmd = "nginx -p " .. self.prefix
+  local ok, code, _, stderr = pl_utils.executeex(cmd)
+  assert(ok and code == 0, "failed to start nginx: " .. stderr)
+  return true
+end
+
+local sleep_step = 0.01
+
+-- stop a dedicate Kong instance for this mock
+function http_mock:stop(no_clean, signal, timeout)
```

Comment:
Is it possible to simplify it with `helpers.wait_pid()`?

Reply:
It does not support other kill signals.